sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -13,13 +13,15 @@
 import java.net.URL;
 import java.nio.charset.Charset;
 import javax.net.ssl.HttpsURLConnection;
+import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * An implementation of the {@link ITransport} interface that sends the events to the Sentry server
  * over HTTP(S) in UTF-8 encoding.
  */
-@Open // TODO: make it final and disable nopen check for testing
+@Open
+@ApiStatus.NonExtendable // only not final because of testing
```

Comment:
Could the tests rely on `ITransport` and make this class final?

Reply:
not actually, because it needs to extend and override the `open` method otherwise `Connection refused` will be thrown on every test

Reply:
the other way would be to create a fake/mock http server, I've used for okhttp, but not for `HttpURLConnection`, will have a look.